fog/test_infra/src/mock_client.rs

```diff
@@ -31,8 +31,11 @@ impl<R: RecoveryDb> FogViewConnection for PassThroughViewClient<R> {
         start_from_block_index: u64,
         search_keys: Vec<Vec<u8>>,
     ) -> Result<QueryResponse, Self::Error> {
-        let (user_events, next_start_from_user_event_id) =
-            self.db.search_user_events(start_from_user_event_id)?;
+        const USER_EVENT_LIMIT: usize = 10_000;
+        let (user_events, next_start_from_user_event_id) = self
+            .db
+            .search_user_events(start_from_user_event_id, USER_EVENT_LIMIT)?;
+        let may_have_more_user_events = user_events.len() >= USER_EVENT_LIMIT;
```

Comment:
It appears that we're returning all the `user_events` in the `query_response` below. If there are `12_000` `user_events`, then we return them all to the client AND we set `may_have_more_user_events` to true. 

However, the logic for `fn search_user_events` has this line: `    .limit(max_num_events as i64)`, and `max_num_events` == `USER_EVENT_LIMIT`. How then can `user_events.len()` ever be > than `USER_EVENT_LIMIT`?